src/hotspot/share/c1/c1_LinearScan.cpp

```diff
@@ -1656,28 +1657,37 @@ void LinearScan::sort_intervals_after_allocation() {
 void LinearScan::allocate_registers() {
   TIME_LINEAR_SCAN(timer_allocate_registers);
 
-  Interval* precolored_cpu_intervals, *not_precolored_cpu_intervals;
-  Interval* precolored_fpu_intervals, *not_precolored_fpu_intervals;
+  Interval* precolored_cpu_intervals = Interval::end(), *not_precolored_cpu_intervals = Interval::end();
+  Interval* precolored_fpu_intervals = Interval::end(), *not_precolored_fpu_intervals = Interval::end();
```

Comment:
Initialization is not necessary as it is already done in `create_unhandled_lists`:
https://github.com/openjdk/jdk17/blob/853d9428e26ba6a6ac703c2b9427f6c6f1bae182/src/hotspot/share/c1/c1_LinearScan.cpp#L1531-L1534